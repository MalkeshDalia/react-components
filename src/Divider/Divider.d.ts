import { CSSProperties, PureComponent } from 'react';
import { Type } from 'tcomb';

export namespace DividerProps {
  type Orientation = 'horizontal' | 'vertical'
  type Size = 'small' | 'medium' | 'large' | 'no-margin'
}

export type DividerProps = {
  orientation?: DividerProps.Orientation,
  style?: CSSProperties,
  size?: DividerProps.Size
}

export default class Divider extends PureComponent<DividerProps> {}

export const Props: {
  [key: string]: Type<any>
}
