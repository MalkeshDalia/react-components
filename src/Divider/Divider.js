import React from 'react';
import { skinnable, props, t } from '../utils';
import cx from 'classnames';

const orientation = t.enums.of(['horizontal', 'vertical'], 'orientation');
const sizeType = t.enums.of(['small', 'medium', 'large', 'no-margin'], 'sizeType');

export const Props = {
  orientation: t.maybe(orientation),
  style: t.maybe(t.Obj),
  size: t.maybe(sizeType)
};

const defaultDividerProps = {
  orientation: 'vertical',
  size: 'small',
  style: {}
};

/** A simple component used to visually divide UI elements
 * @param orientation - vertical | horizontal
 * @param size - size of margins
 */
@skinnable()
@props(Props)
export default class Divider extends React.PureComponent {

  static defaultProps = defaultDividerProps;

  getLocals() {
    return {
      ...this.props
    };
  }

  template({ orientation, style, size }) {
    return (
      <div className={cx('divider', orientation, size)} style={style} />
    );
  }
}
