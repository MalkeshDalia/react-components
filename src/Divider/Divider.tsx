import * as React from 'react';
import { props, t } from '../utils';
import cx from '../utils/classnames';

export namespace DividerProps {
  export type Orientation = 'horizontal' | 'vertical'
  export type Size = 'small' | 'medium' | 'large' | 'no-margin'
};

export type DividerDefaultProps = {
  /** divider orientation (vertical | horizontal) */
  orientation: DividerProps.Orientation,
  /** size of margins */
  size: DividerProps.Size,
  /** an optional style object to pass to top level element of the component */
  style: React.CSSProperties,
};

export type DividerProps = Partial<DividerDefaultProps>;
type DividerDefaultedProps = DividerDefaultProps;

const orientation = t.enums.of(['horizontal', 'vertical'], 'orientation');
const sizeType = t.enums.of(['small', 'medium', 'large', 'no-margin'], 'sizeType');

export const Props = {
  orientation: t.maybe(orientation),
  style: t.maybe(t.Object),
  size: t.maybe(sizeType)
};

/**
 * A simple component used to visually divide UI elements
 */
@props(Props)
export default class Divider extends React.PureComponent<DividerProps> {

  static defaultProps: DividerDefaultProps = {
    orientation: 'vertical',
    size: 'small',
    style: {}
  };

  render() {
    const { orientation, style, size } = this.props as DividerDefaultedProps;
    return (
      <div className={cx('divider', orientation, size)} style={style} />
    );
  }
}
