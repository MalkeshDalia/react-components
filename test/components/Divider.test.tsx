import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Divider from '../../src/Divider';

describe('Divider', () => {

  it('renders correctly vertical', () => {
    const component = renderer.create(
      <Divider orientation='vertical' />
    );
    expect(component).toMatchSnapshot();
  });

  it('renders correctly horizontal', () => {
    const component = renderer.create(
      <Divider orientation='horizontal' />
    );
    expect(component).toMatchSnapshot();
  });

});
