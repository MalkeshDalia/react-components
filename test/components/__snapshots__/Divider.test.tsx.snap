// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider renders correctly horizontal 1`] = `
<div
  className="divider horizontal small"
  style={Object {}}
/>
`;

exports[`Divider renders correctly vertical 1`] = `
<div
  className="divider vertical small"
  style={Object {}}
/>
`;
